engine/api/workflow/dao_run_secret.go

```diff
@@ -0,0 +1,43 @@
+package workflow
+
+import (
+	"context"
+
+	"github.com/go-gorp/gorp"
+
+	"github.com/ovh/cds/engine/api/database/gorpmapping"
+	"github.com/ovh/cds/sdk"
+)
+
+func InsertRunSecret(ctx context.Context, db gorp.SqlExecutor, wrSecret *sdk.WorkflowRunSecret) error {
+	dbData := &dbWorkflowRunSecret{WorkflowRunSecret: *wrSecret}
+	dbData.ID = sdk.UUID()
+	if err := gorpmapping.InsertAndSign(ctx, db, dbData); err != nil {
+		return err
+	}
+	*wrSecret = dbData.WorkflowRunSecret
+	return nil
+}
+
+func loadRunSecretByContext(ctx context.Context, db gorp.SqlExecutor, runID int64, context string) ([]sdk.Variable, error) {
+	var dbSecrets []dbWorkflowRunSecret
+	query := gorpmapping.NewQuery(`SELECT * FROM workflow_run_secret WHERE workflow_run_id = $1 AND context = $2`).Args(runID, context)
+	if err := gorpmapping.GetAll(ctx, db, query, &dbSecrets, gorpmapping.GetOptions.WithDecryption); err != nil {
```

Comment:
I think the `gorpmapping.GetOptions.WithDecryption` should be an option of this function to avoid decrypting all the times
OR
the function should be named `loadRunSecretByContextWithDecryption`

Reply:
done